### PR TITLE
Connection refactoring

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: kademlia/grpc/src/main/scala/fluence/kad/grpc/client/KademliaClient.scala [hunk 1]
```diff
@@ -98,10 +93,17 @@ class KademliaWebsocketClient(connection: IO[WebsocketPipe[WebsocketMessage, Web
...
+      res ← nodeContactCodec.runF[IO](responseBytes)
     } yield res
   }
 }
+
+object KademliaClient {
+  def apply(streamHandler: Connection)(
+    implicit
+    codec: PureCodec[Node[Contact], protobuf.Node],
+    ec: ExecutionContext
```
Comment (RicoGit) [inline kademlia/grpc/src/main/scala/fluence/kad/grpc/client/KademliaClient.scala]: Looks like ExecutionContext isn't used in KademliaClient and might be removed.

File: project/plugins.sbt [hunk 2]
```diff
@@ -16,4 +16,6 @@ addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.9.0")
 
 addSbtPlugin("com.lihaoyi" % "workbench" % "0.4.0")
 
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.0.0-M11")
```
Comment (RicoGit) [inline project/plugins.sbt]: Isn't it break sbt or idea for everyone(without nightly build and so on)? What about scalaJs?

File: scala-multistream/grpc/jvm/src/main/scala/fluence/grpc/ServiceManager.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+package fluence.grpc
+
+import cats.MonadError
+import cats.syntax.flatMap._
+import io.grpc._
+
+import scala.collection.JavaConverters._
+import scala.language.higherKinds
+
+class ServiceManager(services: Map[String, MethodDescriptor[Any, Any]]) {
```
Comment (RicoGit) [inline scala-multistream/grpc/jvm/src/main/scala/fluence/grpc/ServiceManager.scala]: would be nice to have a docs for this class

File: transport/websocket-js/src/main/scala/fluence/transport/websocket/WebsocketConnection.scala [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    WebsocketMessage(service, method, requestId, WebsocketMessage.Response.Payload(ByteString.copyFrom(payload)))
+  }
+
+  override def handle(
+    service: String,
+    method: String,
+    requests: Observable[Array[Byte]]
+  ): IO[Observable[Array[Byte]]] = {
+    connection.map { websocketClient ⇒
+      val requestId = Random.nextLong()
```
Comment (RicoGit) [inline transport/websocket-js/src/main/scala/fluence/transport/websocket/WebsocketConnection.scala]: There is common practice to use UUID for requiest id. 
`UUID.randomUUID().getMostSignificantBits()` returns long value.
Up to you.

File: kademlia/grpc/src/main/scala/fluence/kad/grpc/client/KademliaClient.scala (hunk 1 above)
Comment (DieMyst) [reply kademlia/grpc/src/main/scala/fluence/kad/grpc/client/KademliaClient.scala]: you are right

File: scala-multistream/grpc/jvm/src/main/scala/fluence/grpc/ServiceManager.scala (hunk 3 above)
Comment (DieMyst) [reply scala-multistream/grpc/jvm/src/main/scala/fluence/grpc/ServiceManager.scala]: no, it is only for generation helper classes by sbt command

File: transport/websocket-js/src/main/scala/fluence/transport/websocket/WebsocketConnection.scala (hunk 4 above)
Comment (DieMyst) [reply transport/websocket-js/src/main/scala/fluence/transport/websocket/WebsocketConnection.scala]: there is no UUID in js

**Pull request merged**